datafusion/physical-plan/src/filter.rs

```diff
@@ -91,14 +88,14 @@ impl FilterExec {
 
                 Ok(Self {
                     predicate,
-                    input: input.clone(),
+                    input: Arc::clone(&input),
```

Comment:
This was causing clippy failures on main https://github.com/apache/datafusion/actions/runs/9857745125/job/27217695529

I think due to a logical conflict